test/uuid/ecto/parameterized_type_test.exs

```diff
@@ -0,0 +1,140 @@
+defmodule UUID.Ecto.ParameterizedTypeTest do
+  use ExUnit.Case
+
+  alias UUID.Ecto.Type, as: EctoUUID
+
+  # ----------------------------------------------------------------------------
+  # Setup
+  # ----------------------------------------------------------------------------
+
+  uuids = [
+    UUID.uuid1(),
+    UUID.uuid3(:dns, "mydomain.com"),
+    UUID.uuid4(),
+    UUID.uuid5(:dns, "mydomain.com"),
+    UUID.uuid6(:random_bytes)
+  ]
+
+  infos = Enum.map(uuids, &UUID.info!/1)
+
+  opts = [
+    [type: :uuid1],
+    [type: :uuid3, namespace: :dns, name: "mydomain.com"],
+    [type: :uuid4],
+    [type: :uuid5, namespace: :dns, name: "mydomain.com"],
+    [type: :uuid6, node_type: :random_bytes]
+  ]
+
+  @items Enum.zip([uuids, infos, opts])
+
+  @uuid_null "00000000-0000-0000-0000-000000000000"
+
+  @default_params {:uuid4, []}
+
+  # ----------------------------------------------------------------------------
+  # Tests
+  # ----------------------------------------------------------------------------
+
+  test "type/1" do
+    for {_uuid, _info, opts} <- @items do
+      params = EctoUUID.init(opts)
+      assert EctoUUID.type(params) == :binary_id
```

Comment:
instead of going through implementation details (init/1, etc), I'd consider going through the public interface that Ecto exposes:

```elixir
iex> Ecto.Type.type({:parameterized, UUID.Ecto.Type, type: :uuid6})
:binary_id
```

Same goes for `Ecto.Type.cast/2` etc below.

Reply:
Thanks, that would have been better 😅